### ISSUE TITLE
Parameter $requiredAttributeIds of getUsedProducts() on configurable product type is not used, custom attribute loading does not work

### ISSUE THREAD

**Issue by sippsolutions** [magento/magento2#24483]
### Preconditions (*)
1. Magento Version 2.3.2

### Steps to reproduce (*)
1. Try to load custom attribute's value from \Magento\ConfigurableProduct\Model\Product\Type\Configurable::getUsedProducts()
Parameters are $product, $requiredAttributeIds

### Expected result (*)
1. The attributes in $requiredAttributeIds get loaded and their values are accessible after loading

### Actual result (*)
1. You only have to look at the code to know the parameter does not get used for loading, it just gets used for the cache key!

```
    /**
     * Returns array of sub-products for specified configurable product
     *
     * $requiredAttributeIds - one dimensional array, if provided
     *
     * Result array contains all children for specified configurable product
     *
     * @param  \Magento\Catalog\Model\Product $product
     * @param  array $requiredAttributeIds
     * @return ProductInterface[]
     */
    public function getUsedProducts($product, $requiredAttributeIds = null)
    {
        $metadata = $this->getMetadataPool()->getMetadata(ProductInterface::class);
        $keyParts = [
            __METHOD__,
            $product->getData($metadata->getLinkField()),
            $product->getStoreId(),
            $this->getCustomerSession()->getCustomerGroupId()
        ];
        if ($requiredAttributeIds !== null) {
            sort($requiredAttributeIds);
            $keyParts[] = implode('', $requiredAttributeIds);
        }
        $cacheKey = $this->getUsedProductsCacheKey($keyParts);
        return $this->loadUsedProducts($product, $cacheKey);
    }
```

**Comment by ghost** [comment]
@magento I am  working on it.

**Comment by ghost** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @prabhatrawat-webkul. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @prabhatrawat-webkul, here is your [Magento instance](https://i-24483-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-24483-2-3-develop.instances.magento-community.engineering/admin](https://i-24483-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20142` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @sippsolutions. Thank you for your report. 
The issue has been fixed in magento/magento2#24875 by @lfolco in 2.3-develop branch
Related commit(s):
 - [8ca53094733442f1e06908592979c405ed151607](https://github.com/magento/magento2/commit/8ca53094733442f1e06908592979c405ed151607)
 - [b4dfa30d5e0cb22f65e6f57a1e56e71835b0df27](https://github.com/magento/magento2/commit/b4dfa30d5e0cb22f65e6f57a1e56e71835b0df27)
 - [567a6e45e8a245ee6b794097a404ae6b918852b1](https://github.com/magento/magento2/commit/567a6e45e8a245ee6b794097a404ae6b918852b1)
 - [6e401d6f8f5b49e975c46e361bead0cf9dc15054](https://github.com/magento/magento2/commit/6e401d6f8f5b49e975c46e361bead0cf9dc15054)
 - [7f7e840338f1bb7dbd7fa78b1dba9ac0173ad7fd](https://github.com/magento/magento2/commit/7f7e840338f1bb7dbd7fa78b1dba9ac0173ad7fd)
 - [c70d80e4fa3157ff492cf3d1e9ceddb9af687b8d](https://github.com/magento/magento2/commit/c70d80e4fa3157ff492cf3d1e9ceddb9af687b8d)
 - [3d12cb4a57f92508ef23010b7df0698cb04fe6a1](https://github.com/magento/magento2/commit/3d12cb4a57f92508ef23010b7df0698cb04fe6a1)
 - [5a6deee9dfe347ab7c668afe97bbe3161f166021](https://github.com/magento/magento2/commit/5a6deee9dfe347ab7c668afe97bbe3161f166021)
 - [7ca1f17d2a7dc6e42f7f100b24fab976936542fe](https://github.com/magento/magento2/commit/7ca1f17d2a7dc6e42f7f100b24fab976936542fe)
 - [d73f413ad225f3d22bc759d4a5d2e426ee989fde](https://github.com/magento/magento2/commit/d73f413ad225f3d22bc759d4a5d2e426ee989fde)
 - [988fb8b470ea988a005534ab96003234db824ab1](https://github.com/magento/magento2/commit/988fb8b470ea988a005534ab96003234db824ab1)
 - [a436610b577a187d14c879ed693359f09a3383a7](https://github.com/magento/magento2/commit/a436610b577a187d14c879ed693359f09a3383a7)
 - [f4cbd32d8013fc653ed5c5fafc3baf167ec01f04](https://github.com/magento/magento2/commit/f4cbd32d8013fc653ed5c5fafc3baf167ec01f04)

The fix will be available with the upcoming 2.3.4 release.
